apps/dashboard/src/components/subscribers/locale-select.tsx

```diff
@@ -0,0 +1,71 @@
+import { locales } from '@/utils/locales';
+import { RiEarthLine } from 'react-icons/ri';
+import { CountryFlags } from '../icons/country-flags';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '../primitives/select';
+import TruncatedText from '../truncated-text';
+
+export function LocaleSelect({
```

Comment:
I'm not sure why but this component is working super slowly for me locally :/ 

https://github.com/user-attachments/assets/80d5555c-7ed2-4ec9-9200-9cd5d458b82e

Reply:
It's coz of approx 500 options list items
There is an improvement for this and other improvements like introducing `/globals` API endpoint which I will do in a separate PR